i18n-helpers/src/lib.rs

```diff
@@ -477,6 +477,54 @@ fn heuristic_codeblock<'a>(
     Ok((groups, ctx))
 }
 
+/// Special Admonish "codeblock" - extract the body content for translation
+/// If a title is present, also extract that for translation
+/// Note this is from mdbook-admonish: https://github.com/tommilligan/mdbook-admonish
+/// This assumes it is called by `parse_codeblock()` when `is_admonish() == true`
+/// Thus, it does not repeat many of the identical checks to save some FLOPS
+fn admonish_codeblock<'a>(
+    events: &'a [(usize, Event<'_>)],
+    mut ctx: GroupingContext,
+) -> Result<(Vec<Group<'a>>, GroupingContext), CmarkError> {
+    // Handle the entire block as a single translatable unit
+    // The translate_events function will take care of matching the proper translations
+    // for both the title and body content based on the PO file entries
+    let groups = vec![Group::Translate {
+        events: events.into(),
+        comment: std::mem::take(&mut ctx.comments).join(" "),
+    }];
+
+    Ok((groups, ctx))
+}
+
+/// Check if the code block is an admonish block
+fn is_admonish(events: &[(usize, Event<'_>)]) -> bool {
+    const ADMONISH_CODEBLOCK_NAME: &str = "admonish";
+
+    // Ensure we have at least two events
+    if events.len() < 2 {
+        return false;
+    }
+
+    // ensure we are in a proper code block
+    let (_, Event::End(TagEnd::CodeBlock)) = &events[events.len() - 1] else {
+        return false;
+    };
+
+    // pull the info_string (aka language specifier) out of the code block
+    // (the string after the ```)
+    let (_, Event::Start(Tag::CodeBlock(CodeBlockKind::Fenced(info_string)))) = &events[0] else {
+        return false;
+    };
+
+    // Check if the language specifier contains "admonish"
+    info_string
+        .split(',')
+        .next()
+        .unwrap_or_default()
+        .contains(ADMONISH_CODEBLOCK_NAME)
```

Comment:
Would this do more or less the same:

```suggestion
    match events {
        [(_, Event::Start(Tag::CodeBlock(CodeBlockKind::Fenced(info_string)))), .., (_, Event::End(TagEnd::CodeBlock))] => {
            info_string.split(' ').next() == Some(ADMONISH_CODEBLOCK_NAME)
        }
        _ => false,
    }
```

This changes the splitting to use `' '` and uses a slice pattern to assert the overall structure in one go.

I skimmed the `mdbook-admonish` code and found [this code](https://github.com/tommilligan/mdbook-admonish/blob/21d86546ffbcad83108f163c5805737001c0c2a3/src/config/mod.rs#L19-L32), which seems to split on `' '` instead.

Down the road, I think we can avoid special-casing `admonish` and instead have a configurable list of languages (from the info string) which should be fully extracted. We can also have a list of strings that should be unconditionally ignored. But that is for another PR :smile: